alf/environments/suite_dmc.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) 2021 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import alf
+from alf.environments.alf_environment import AlfEnvironment
+from alf.environments.suite_gym import wrap_env
+from unittest.mock import Mock
+
+try:
+    import dmc2gym
+except ImportError:
+    # create 'dmc2gym' as a mock to not break python argument type hints
+    dmc2gym = Mock()
+
+
+def is_available():
+    return not isinstance(dmc2gym, Mock)
+
+
+class AlfEnvironmentDMC2GYMWrapper(AlfEnvironment):
```

Comment:
If you subclass from AlfEnvironmentBaseWrapper, you can remove a lot of functions already implemented in AlfEnvironmentBaseWrapper